src/resolve/index.ts

```diff
@@ -6,17 +6,73 @@ import {Got} from '../network/got'
 import {Package} from '../types'
 import {LoggedPkg} from 'pnpm-logger'
 
-export type ResolveResult = {
+export type ResolutionBase = {
   id: string,
+}
+
+/**
+ * npm registry hosted package
+ */
+export type PackageResolution = ResolutionBase & {
+  type: 'package',
+  tarball: string,
+  shasum?: string,
   pkg?: Package,
+}
```

Comment:
Actually I'm not sure we need a separate type for packages from npm, probably this could be just `TarballResoluton`. We could move `pkg` attr to `ResolutionBase` — as I understand it is there only for caching purposes.